automation/check-patch.e2e-multus-functests.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash -xe
+
+# This script should be able to execute multus
+# functional tests against Kubernetes cluster with
+# CNAO built with latest changes, on any
+# environment with basic dependencies listed in
+# check-patch.packages installed and docker running.
+#
+# yum -y install automation/check-patch.packages
+# automation/check-patch.e2e-multus-functests.sh
+
+teardown() {
+    make cluster-down
+    rm -rf "${TMP_COMPONENT_PATH}"
+}
+
+function __get-tools() {
+    if [ ! -d bin ]; then
+        mkdir bin
+    fi
+
+    echo "install jq"
+    curl -Lo ./bin/jq https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64
+    chmod +x ./bin/jq
+}
+
+function __prepare-test-environment() {
+    __get-tools
+
+    echo "Deploy cni-plugins pod"
+    CNI_PLUGIN_YAML="cni-install.yml"
+    kubectl create -f ${CNI_PLUGIN_YAML}
+    kubectl -n kube-system wait --for=condition=ready -l name=cni-plugins pod --timeout=300s
+
+    TEST_YAML="*macvlan1.yml"
+    # matching the test nodes to the CNAO cluster nodes
+    sed -i 's/kind-worker$/node01/g' ${TEST_YAML}
+    sed -i 's/kind-worker2$/node02/g' ${TEST_YAML}
+
+    echo "Set eth1 NIC to up using NodeNetworkConfigurationPolicy"
+    cat <<EOF | kubectl apply -f -
+apiVersion: nmstate.io/v1beta1
+kind: NodeNetworkConfigurationPolicy
+metadata:
+  name: eth1-policy
+spec:
+  desiredState:
+    interfaces:
+    - name: eth1
+      description: Configuring eth1 on all nodes
+      type: ethernet
+      state: up
+      ipv4:
+        dhcp: true
+        enabled: true
+EOF
+}
+
+function __remove-test-manifests() {
+    echo "Remove test manifests"
+    kubectl delete -f ${TEST_YAML}
+    kubectl delete -f ${CNI_PLUGIN_YAML}
+}
+
+main() {
+    # Setup CNAO and artifacts temp directory
+    source automation/check-patch.setup.sh
+    cd ${TMP_PROJECT_PATH}
```

Comment:
where TMP_PROJECT_PATH is declared?
maybe u want set -u , so it will fail if its not declared?

its ok i guess that there is TMP_COMPONENT_PATH and TMP_PROJECT_PATH right?

Reply:
> where TMP_PROJECT_PATH is declared?

inside automation/check-patch.setup.sh, which is used for all automation scripts.

> maybe u want set -u , so it will fail if its not declared?

will do

> 
> its ok i guess that there is TMP_COMPONENT_PATH and TMP_PROJECT_PATH right?

yes, the former refers multus/kmp/... and the latter refers to cnao itself